bitbake.el

```diff
@@ -740,22 +740,38 @@ For detail, see `comment-dwim'."
   (set (make-local-variable 'indent-line-function) 'bitbake-indent-line)
   (define-key bitbake-mode-map [remap comment-dwim] 'bitbake-comment-dwim))
 
+(defvar bitbake-shell-regex "^\\(fakeroot[[:space:]]*\\)?\\([a-zA-Z0-9\-_+.${}/~]+\\)[[:space:]]*([[:space:]]*)[[:space:]]*{")
+(defvar bitbake-python-regex "^\\(fakeroot[[:space:]]*\\)?python[[:space:]]*\\([a-zA-Z0-9\-_+.${}/~]+\\)?[[:space:]]*([[:space:]]*)[[:space:]]*{")
+
+(defun bitbake-shell-front-verify ()
+  (not (string-match bitbake-python-regex (match-string 0))))
+
 (mmm-add-classes
- '((bitbake-shell
+ `((bitbake-shell
     :submode shell-script-mode
     :delimiter-mode nil
     :case-fold-search nil
-    :front "^\\(fakeroot *\\)?\\([a-zA-Z0-9\-_+.${}/~]+\\) *( *) *{"
+    :front ,bitbake-shell-regex
+    :front-verify bitbake-shell-front-verify
     :back "^}")
    (bitbake-python
     :submode python-mode
     :delimiter-mode nil
     :case-fold-search nil
-    :front "^\\(fakeroot *\\)?python *\\([a-zA-Z0-9\-_+.${}/~]+\\) *( *) *{"
-    :back "^}")))
+    :front ,bitbake-python-regex
+    :back "^}")
+   (bitbake-python2
+    :submode python-mode
+    :delimiter-mode nil
+    :case-fold-search nil
+    :front "^def +[a-zA-Z0-9_]+[[:space:]]*([[:space:]a-zA-Z0-9_,]*)[[:space:]]*:"
```

Comment:
Maybe also use a variable here as for the two previous classes.

Reply:
Done.

Reply:
Also renamed the bitbake-python2 class to bitbake-python-def, since that was a bit more descriptive.